ccpp/data/GFS_typedefs.F90

```diff
@@ -4165,6 +4175,7 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%ivegsrc          = ivegsrc
     Model%isot             = isot
     Model%use_ufo          = use_ufo
+    Model%exticeden        = exticeden
```

Comment:
I put the following guard rail in my branch:`if (Model%exticeden .and. &
      (Model%imp_physics /= Model%imp_physics_gfdl .and. Model%imp_physics /= Model%imp_physics_thompson .and. &
       Model%imp_physics /= Model%imp_physics_nssl )) then`
      `!see GFS_MP_generic_post.F90; exticeden is only compatible with GFDL, Thompson, or NSSL MP `
      `print *,' Using exticeden = T is only valid when using GFDL, Thompson, or NSSL microphysics.'`
      `stop`
    `end if`

Reply:
I put this check in here because the calculation of rhonewsn in GFS_MP_generic_post is within an IF statement for these 3 MP schemes. If we don't have a check like this, it would be possible to specify exticeden = T in the namelist for a suite using a different microphysics and the LSMs would bypass their internal calculation of frozen precipitation density in favor of a bogus/initialized-only value of rhonewsn.

Reply:
You should also replace line 4200 with the following:
`if (Model%lsm==Model%lsm_noahmp .and. Model%exticeden .and. iopt_snf == 4) then`
 `      Model%iopt_snf         = 5`
   ` else`
     ` Model%iopt_snf         = iopt_snf`
    `end if`

Reply:
When NoahMP is used, the iopt_snf==5 option is used to cause the scheme to use the externally-calculated value of surface frozen precipitation ice density.

Reply:
You can view the changes that I'm talking about in GFS_typedefs.F90 here: https://github.com/ericaligo-NOAA/fv3atm/pull/1/files

Reply:
I agree with all of your changes. See my earlier comments and concern about rhosnfall.